content/workflow.md

```diff
@@ -0,0 +1,406 @@
+
+```java
+type LibraryWork {
+    a: String
+    previous: Json
+    session: WorkflowSession,
+    params: Json
+}
+
+type Query {
+    stepa: String@tan(type:Groovy, inline: "return 'Welcome to Library!'")
+    stepb: Json@tan(type:Groovy,inline:"""
+      return gql(\"""
+        query{
+                find (
+                type: Author, 
+                arcql: "*")
+                {
+                    edges {
+                    cursor
+                    node {
+                        ... onAuthor {
+                        hypi {
+                           id
+                        }
+                        name
+                        }
+                    }
+                  }
+               }           
+           }  
+    \"""
+        )
+    """)  
+    stepc(a:String): Json@tan(type:Groovy, inline: """
+      return  gql(\"""
+        query {
+            get(type: Author, id: "$a"){
+                ... onAuthor {
+                    hypi{
+                        id
+                     }
+                    name
+                booklist{
+                    title
+                    price
+                    authorid
+                }
+            }
+        }
+    }\"""
+    )
+    """)
+    stepd(b: String): [String!] @tan(type:Groovy, inline: """
+    def results =  gql(\"""
+    query {
+        find (
+            type: Book, 
+            arcql: "title = '$b'"
+         ) {
+                edges {
+                    cursor
+                    node {
+                    ... onBook {
+                        hypi {
+                        id
+                        }
+                    }
+                }
+              }
+            }
+          }
+   \"""
+   ) 
+   return results.data.find.edges.stream()
+   .map({o -> o.node.hypi.id})
+   .collect(java.util.stream.Collectors.toList())  
+""")  
+   stepf(a: String, previous: String, session: WorkflowSessionInput, 
+   params: Json): 
+   LibraryWork@tan(type:Groovy, inline: """
+   return [
+    "a": a,
+    "previous": previous,
+    "session": session,
+    "params": params
+  ]
+""")
+   BorrowBook(a: String,b: String): LibraryWork@workflow(name: "BorrowBook")
+}
+
+type Mutation {
+  # It is entirely up to the author to decide if their custom function 
+  is mutation or not
+  stepe(previous:\[String!\]): Json@tan(type:Groovy, inline: """
+      return gql(\"""
+        mutation {
+        upsert(
+            values: { 
+                Book: [
+                { 
+                    hypi:{
+                        id:"${previous[0]}"
+                    }
+                    available: false
+                    }
+                 ] 
+            }) {
+                id
+            }
+        }\"""
+    )
+""")
+}
+```
+
+> Important notes to understand the schema!
+
+The first step can have any parameters you want. For the other steps, some rules define how the system maps parameters to the function in these steps.
```

Comment:
`rules define` should be `rules defined`

Reply:
are defined?